python/cuml/tests/test_linear_model.py

```diff
@@ -193,10 +204,38 @@ def test_linear_regression_single_column():
         model.fit(cp.random.rand(46341), cp.random.rand(46341))
 
 
-@pytest.mark.parametrize("datatype", [np.float32, np.float64])
-def test_linear_regression_model_default(datatype):
-
-    X_train, X_test, y_train, y_test = small_regression_dataset(datatype)
+@given(
+    split_datasets(
+        regression_datasets(
+            # Two assumptions required for cuml.LinearRegression:
+            n_samples=st.integers(
+                min_value=20, max_value=200
+            ),  # assuming min(train_size)=0.1
+            dtypes=floating_dtypes(sizes=(32, 64)),
+        )
+    )
+)
+@example(small_regression_dataset(np.float32))
+@example(small_regression_dataset(np.float64))
+@settings(
+    deadline=5000, max_examples=20
+)  # TODO: re-evaluate max_examples after benchmarking
+def test_linear_regression_model_default(dataset):
+
+    X_train, X_test, y_train, y_test = dataset
+    n_rows, n_cols = X_train.shape
+
+    ## Required assumptions:
+    # sklinearRegression:
+    assume(n_cols >= 1)
+    assume((X_train > 0).any())
+    assume((y_train > 0).any())
+    assume(np.isfinite(X_train).all())
+    assume(np.isfinite(y_train).all())
+    # cuml.LinearRegression:
```

Comment:
Really like that you documented which assumptions go with which algorithm! I'm a bit surprised that cuML does not accept a single row while sklearn does here. Is it obvious to you why, and is that a limitation that we might be able to eliminate?

Reply:
The estimator fit function is guarded against this here: https://github.com/csadorf/cuml/blob/fe6d2045c191d38f7ce36c9183a5bf2aba042110/python/cuml/linear_model/linear_regression.pyx#L263-L265

It is not immediatley obvious to me why this is needed, but I can look into it.

Reply:
No need! We should just make sure it's documented, but that's tangential to this PR. Thanks!